enterprise/cypher/slotted-runtime/src/main/scala/org/neo4j/cypher/internal/runtime/slotted/pipes/DistinctSlottedPrimitivePipe.scala

```diff
@@ -90,3 +93,29 @@ case class DistinctSlottedPrimitivePipe(source: Pipe,
     keys
   }
 }
+
+/**
+  * This little class is here to make sure we have the expected behaviour of our primitive arrays.
+  * In the JVM, long[] are do not have reasonable hashcode or equal()
+  */
+class Key(val inner: Array[Long]) {
+
+  private val _hashCode = util.Arrays.hashCode(inner)
+
+  override def equals(other: Any): Boolean = {
+    if ( other == null || getClass() != other.getClass() )
```

Comment:
I think this is equivalent with
```
other match {
   case k: Key => _hashCode == otherKey._hashCode && util.Arrays.equals(inner, otherKey.inner)
   case _ => false
}
```

I don't think that will create any lambdas just do an instance of check, I'm always scared about using return.